File: app/src/main/java/com/paintology/lite/trace/drawing/Activity/favourite/DrawingAdapter.java
package com.paintology.lite.trace.drawing.Activity.favourite;

import static com.paintology.lite.trace.drawing.Activity.utils.ExtensionsKt.openActivity;

import android.annotation.SuppressLint;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.PopupMenu;

import androidx.appcompat.widget.AppCompatButton;
import androidx.appcompat.widget.AppCompatImageView;
import androidx.appcompat.widget.AppCompatTextView;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.google.android.material.imageview.ShapeableImageView;
import com.paintology.lite.trace.drawing.Activity.gallery_activity.views.activities.DrawingViewActivity;
import com.paintology.lite.trace.drawing.Activity.gallery_activity.views.fragment.GalleryTutorailsFragment;
import com.paintology.lite.trace.drawing.Activity.user_pogress.helper.TutorialUtils;
import com.paintology.lite.trace.drawing.Activity.utils.ExtensionsKt;
import com.paintology.lite.trace.drawing.DashboardScreen.CategoryActivity;
import com.paintology.lite.trace.drawing.R;
import com.paintology.lite.trace.drawing.util.ContextKt;
import com.paintology.lite.trace.drawing.util.FireUtils;
import com.paintology.lite.trace.drawing.util.StringConstants;
import com.paintology.lite.trace.drawing.util.events.RefreshFavoriteEvent;

import org.greenrobot.eventbus.EventBus;

import java.util.List;

public class DrawingAdapter extends RecyclerView.Adapter<DrawingAdapter.DrawingViewHolder> {
    private Context context;
    StringConstants constants = new StringConstants();
    private List<com.paintology.lite.trace.drawing.Activity.gallery_activity.model.new_models.NewDrawing> drawings;

    public DrawingAdapter(Context context, List<com.paintology.lite.trace.drawing.Activity.gallery_activity.model.new_models.NewDrawing> drawings) {
        this.context = context;
        this.drawings = drawings;
    }

    public static class DrawingViewHolder extends RecyclerView.ViewHolder {

        public ShapeableImageView imgThumbnail;
        public AppCompatTextView tvTutorialContent;
        public AppCompatTextView tvComments;
        public AppCompatButton btnDrawing;
        public AppCompatTextView tvLikes;
        public AppCompatTextView tvName;
        public me.zhanghai.android.materialratingbar.MaterialRatingBar RatingBar;
        public AppCompatImageView appCompatImageView7;
        public AppCompatImageView imgMenu12;
        // Add other views

        public DrawingViewHolder(View itemView) {
            super(itemView);

            imgThumbnail = itemView.findViewById(R.id.imgThumbnail);
            imgMenu12 = itemView.findViewById(R.id.imgMenu12);
            tvTutorialContent = itemView.findViewById(R.id.tvTutorialContent);
            btnDrawing = itemView.findViewById(R.id.btnDrawing);
            tvComments = itemView.findViewById(R.id.tvComments);
            RatingBar = itemView.findViewById(R.id.RatingBar);
            tvLikes = itemView.findViewById(R.id.tvLikes);
            tvName = itemView.findViewById(R.id.tvName);
            appCompatImageView7 = itemView.findViewById(R.id.appCompatImageView7);

        }
    }

    @Override
    public DrawingViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.layout_gallery_tutorial_item, parent, false);
        return new DrawingViewHolder(view);
    }

    @Override
    public void onBindViewHolder(DrawingViewHolder holder, @SuppressLint("RecyclerView") int position) {
        com.paintology.lite.trace.drawing.Activity.gallery_activity.model.new_models.NewDrawing drawing = drawings.get(position);

        holder.tvTutorialContent.setText(drawing.getDescription());

        holder.tvName.setText(drawing.getAuthor().getName());
        holder.btnDrawing.setText(drawing.getTitle());

        holder.imgMenu12.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ShowMenu(holder.getAdapterPosition(), v);

            }
        });

        holder.imgThumbnail.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Bundle bundle = new Bundle();
                bundle.putString("post_id", drawings.get(position).getId());
                ContextKt.sendUserEventWithParam(context, StringConstants.favorites_gallery_open, bundle);
                ExtensionsKt.startNewDrawingActivity(context, drawings.get(position), DrawingViewActivity.class, false);
            }
        });

//        holder.RatingBar.setNumStars(drawing.getStatistic().getRatings().intValue());

        holder.tvLikes.setText(String.valueOf(drawing.getStatistic().getLikes().intValue()));
        holder.tvComments.setText(String.valueOf(drawing.getStatistic().getComments().intValue()));
//        holder.appCompatImageView7.setText(drawing.getStatistic().getComments());

        Glide.with(context)
                .load(drawing.getImages().getContent()) // image url
                .placeholder(R.drawable.img_cat_dummy) // any placeholder to load at start// any image in case of error
                .override(200, 200) // resizing
                .centerCrop()
                .into(holder.imgThumbnail);

        // Set other data
    }

    @Override
    public int getItemCount() {
        return drawings.size();
    }

    public void setDrawings(List<com.paintology.lite.trace.drawing.Activity.gallery_activity.model.new_models.NewDrawing> drawings) {
        this.drawings = drawings;
        notifyDataSetChanged();
    }

    private void ShowMenu(int position, View view) {
        PopupMenu popup = new PopupMenu(context, view);
        //Inflating the Popup using xml file
        popup.getMenuInflater().inflate(R.menu.gallery_tut_fav_remove_menu, popup.getMenu());

        //registering popup with OnMenuItemClickListener
        popup.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
            public boolean onMenuItemClick(MenuItem item) {

                if (item.getItemId() == R.id.openItem1) {
                    ExtensionsKt.startNewDrawingActivity(context, drawings.get(position), DrawingViewActivity.class, false);
                    return true;
                } else if (item.getItemId() == R.id.doTutorialItem1) {

                    if (drawings.get(position).getMetadata().getTutorialId().isEmpty()) {
                        openActivity(context, CategoryActivity.class);
                    } else {
                        FireUtils.showProgressDialog(
                                context,
                                context.getResources().getString(R.string.please_wait)
                        );
                        new TutorialUtils(context).parseTutorial(drawings.get(position).getMetadata().getTutorialId());
                    }

                    return true;
                } else if (item.getItemId() == R.id.shareItem11) {
                    GalleryTutorailsFragment fragment = new GalleryTutorailsFragment();
                    fragment.loadImageAndSave(drawings.get(position).getImages().getContent(), context, drawings.get(position));
                    return true;
                } else if (item.getItemId() == R.id.RemoveItem1) {

                    DrawingRepository repository = new DrawingRepository(context);
                    repository.RemoveDrawing(drawings.get(position).getId());
                    drawings.remove(position);
                    notifyItemRemoved(position);
                    EventBus.getDefault().post(new RefreshFavoriteEvent(1));
                    return true;
                }
                return false;

            }
        });

        popup.show();
    }


}

File: app/src/main/java/com/paintology/lite/trace/drawing/Activity/user_pogress/adapter/DrawingAdapter.kt
package com.paintology.lite.trace.drawing.Activity.user_pogress.adapter

import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.paintology.lite.trace.drawing.Activity.gallery_activity.views.activities.GalleryActivity
import com.paintology.lite.trace.drawing.Activity.user_pogress.listener.onTutorialClickListener
import com.paintology.lite.trace.drawing.databinding.LayoutUserDrawingBinding
import com.paintology.lite.trace.drawing.databinding.LayoutViewAllBinding
import com.squareup.picasso.Picasso

class DrawingAdapter(
    val context: Activity,
    private val drawingsList: List<Any?> = mutableListOf(),
    private val onTutorialClickListener: onTutorialClickListener
) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        if (viewType == 0) {
            val view =
                LayoutUserDrawingBinding.inflate(
                    LayoutInflater.from(parent.context),
                    parent,
                    false
                )
            return ViewHolder(view)
        } else {
            val view =
                LayoutViewAllBinding.inflate(
                    LayoutInflater.from(parent.context),
                    parent,
                    false
                )
            return ViewAllHolder(view)
        }
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if (holder is ViewHolder) {
            val item = drawingsList.get(position) as HashMap<*, *>
            val images = item.get("images") as HashMap<*, *>
            if (images.containsKey("content")) {
                Picasso.get().load(Uri.parse(images["content"].toString()))
                    .into(holder.binding.imgThumbnail)
            }
            val stats = item.get("statistic") as HashMap<*, *>
            holder.binding.apply {
                btnDrawing.text = item.get("title").toString()
                tvTutorialContent.text = item.get("description").toString()
                tvLikes.text = stats.get("likes").toString()
                tvComments.text = stats.get("comments").toString()
            }
            holder.itemView.setOnClickListener {
                onTutorialClickListener.onTutorialClick(position, item)
            }
        } else {
            holder.itemView.setOnClickListener {
                context.startActivity(Intent(context, GalleryActivity::class.java))
            }
        }
    }

    override fun getItemCount(): Int {
        return drawingsList.size
    }

    override fun getItemViewType(position: Int): Int {
        return if (drawingsList.get(position) == null) 1 else 0
    }

    inner class ViewHolder(val binding: LayoutUserDrawingBinding) :
        RecyclerView.ViewHolder(binding.root)

    inner class ViewAllHolder(val binding: LayoutViewAllBinding) :
        RecyclerView.ViewHolder(binding.root)
}